pyspecdata/lmfitdata.py

```diff
@@ -100,40 +100,16 @@ def functional_form(self, this_expr):
         )
         self.fit_axis = list(self.fit_axis)[0]
         # }}}
-        args = self.parameter_symbols + [str(*this_axis)]
-        self.fitfunc_multiarg = sp.lambdify(
-            args,
-            self.expression,
-            modules=[{"ImmutableMatrix": np.ndarray}, "numpy", "scipy"],
-        )
         self.fitfunc_multiarg_v2 = sp.lambdify(
             self.variable_symbols + self.parameter_symbols,
             self.expression,
             modules=[{"ImmutableMatrix": np.ndarray}, "numpy", "scipy"],
         )
 
-        def fn(p, x):
-            p = self.add_inactive_p(p)
-            assert len(p) == len(
-                self.parameter_names
-            ), "length of parameter passed to fitfunc doesnt match number of symbolic parameters"
-            return self.fitfunc_multiarg(*tuple(list(p) + [x]))
-
-        self.fitfunc = fn
```

Comment:
this add_inactive stuff is obsolete because lmfit has a mechanism for fixing parameters (which was the purpose of the "inactive parameter"

also fitfund *should not be used, since it is not how the residual calculates the fit function*

Reply:
note that this change will **break** eval -- I know this -- do not fix eval by rolling back this change.

Reply:
self.fitfunc is used later on line 256?